src/Obsidian/Cache.ts

```diff
@@ -91,6 +92,20 @@ export function getTasksFromFileContent2(
                     ),
                     fallbackDate: dateFromFileName.value,
                 });
+
+                if (task !== null) {
+                    const parentTask: Task | null = line2Task.get(listItem.parent) ?? null;
+                    task = new Task({
+                        ...task,
+                        parent: parentTask,
+                    });
```

Comment:
if `parentTask` is null, this line is perhaps unnecessary... For clarity, perhaps move it inside the `if` block on line 103?

Reply:
good idea, fixed that